src/librustc/mir/mod.rs

```diff
@@ -225,13 +224,19 @@ impl<'tcx> Body<'tcx> {
     }
 
     #[inline]
-    pub fn predecessors(&self) -> MappedReadGuard<'_, IndexVec<BasicBlock, Vec<BasicBlock>>> {
-        self.cache.predecessors(self)
+    pub fn predecessors_ref(&self) -> &IndexVec<BasicBlock, Vec<BasicBlock>> {
+        self.cache.predecessors_ref()
     }
 
     #[inline]
-    pub fn predecessors_for(&self, bb: BasicBlock) -> MappedReadGuard<'_, Vec<BasicBlock>> {
-        MappedReadGuard::map(self.predecessors(), |p| &p[bb])
+    pub fn predecessors_mut(&mut self) -> &mut IndexVec<BasicBlock, Vec<BasicBlock>> {
```

Comment:
I don't think any use-site uses the result mutably

```suggestion
    pub fn predecessors_mut(&mut self) -> &IndexVec<BasicBlock, Vec<BasicBlock>> {
```

Reply:
This function isn't for mutating the `predecessors`, so maybe just call this one `predecessors` and call the `predecessors_ref` `unwrap_predecessors`